node/cn/peer.go

```diff
@@ -993,28 +993,51 @@ func (p *multiChannelPeer) UpdateRWImplementationVersion() {
 
 func (p *multiChannelPeer) ReadMsg(rw p2p.MsgReadWriter, connectionOrder int, errCh chan<- error, wg *sync.WaitGroup, closed <-chan struct{}) {
 	defer wg.Done()
+
+	readMsgCh := make(chan struct {
+		p2p.Msg
+		error
+	}, channelSizePerPeer)
+	go func() {
+		for {
+			// TODO-klaytn: check 30-second timeout works
+			msg, err := rw.ReadMsg()
+			readMsgCh <- struct {
+				p2p.Msg
+				error
+			}{msg, err}
+		}
+	}()
+
 	for {
-		msg, err := rw.ReadMsg()
+		var (
+			msg p2p.Msg
+			err error
+		)
+		select {
+		case pair := <-readMsgCh:
+			msg, err = pair.Msg, pair.error
+		case <-closed:
+			return
```

Comment:
```suggestion
		case <-closed:
		        close(readMsgCh)
			return
```

How about adding a close channel logic when for loop is closed?

Reply:
Oh you added a close logic at https://github.com/klaytn/klaytn/pull/1389/commits/8f988f8458dc0ea3ed7c3d0a79de1eee65233605.

Reply:
thanks for the comment, i changed the child loop to receive close signal when closing `closed` channel.